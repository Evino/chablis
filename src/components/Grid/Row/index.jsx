import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { breakpoints, sizes } from '../../../styles/variables';


const RowDiv = styled.div`
  width: 100%;
  display: flex;
  box-sizing: border-box;
  flex-flow: row wrap;
  align-items: ${props => props.align};

  @media(min-width: ${breakpoints.EXTRASMALL}) {
    padding: 0 ${sizes.HALF};
  }

  ${props => props.styles}
`;

export default class Row extends PureComponent {
  static propTypes = {
    children: PropTypes.node.isRequired,
    styles: PropTypes.string,
    align: PropTypes.oneOf([
      'flex-start',
      'center',
      'flex-end',
      'baseline',
      'stretch',
    ]),
  }

  static defaultProps = {
    align: 'flex-start',
  }

  render() {
    return (
      <RowDiv {...this.props}>{this.props.children}</RowDiv>
    );
  }
}
