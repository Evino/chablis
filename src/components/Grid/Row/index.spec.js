import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Row from './';

const element = <Row><span>Subcomponent</span></Row>;
const component = mount(element);

describe('Row', () => {
  it('should render a row div', () => {
    const tree = renderer
      .create(element)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render children properly', () => {
    const child = component.find('span');
    expect(child).toHaveLength(1);
  });

  it('should render with default alignment', () => {
    expect(component.prop('align')).toBe('flex-start');
  });

  it('should render with custom aligment', () => {
    component.setProps({ align: 'center' });
    expect(component.prop('align')).toBe('center');
  });
});
