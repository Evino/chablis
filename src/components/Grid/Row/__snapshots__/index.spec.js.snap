// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row should render a row div 1`] = `
<div
  className="sc-bdVaJa ldQJtT"
>
  <span>
    Subcomponent
  </span>
</div>
`;
